logger/logger.go

```diff
@@ -200,8 +200,11 @@ type zapLogger[T zaputil.Encoder[T]] struct {
 	sampler   *zaputil.Sampler
 }
 
-func NewZapLogger(conf *Config, opts ...ZapLoggerOption) (ZapLogger, error) {
-	zap := zap.New(nil).WithOptions(zap.AddCaller(), zap.AddStacktrace(zap.ErrorLevel)).Sugar()
+func NewZap(log *zap.Logger, conf *Config, opts ...ZapLoggerOption) (ZapLogger, error) {
```

Comment:
The two API names are too similar and I think it'll be confusing.

how about having the core method to be `FromZapLogger`, and then `NewZapLogger` would create an instance of zap and passing it in